dandiapi/api/asset_paths.py

```diff
@@ -154,3 +155,19 @@ def add_version_asset_paths(version: Version):
     """Add every asset from a version."""
     for asset in version.assets.iterator():
         add_asset_paths(asset, version)
+
+
+def add_zarr_paths(zarr: ZarrArchive):
+    """Add all asset paths that are associated with a zarr."""
+    # Only act on draft assets/versions
+    for asset in zarr.assets.filter(published=False):
+        for version in asset.versions.filter(version='draft'):
+            add_asset_paths(asset, version)
+
+
+def delete_zarr_paths(zarr: ZarrArchive):
+    """Remove all asset paths that are associated with a zarr."""
+    # Only act on draft assets/versions
+    for asset in zarr.assets.filter(published=False):
+        for version in asset.versions.filter(version='draft'):
+            delete_asset_paths(asset, version)
```

Comment:
Would it be worth using `.iterator()` for all the querysets here? As long as these functions are only called inside tasks, I don't see any downsides to doing so. In normal operations, I don't think we'd ever have a `ZarrArchive` with enough `Assets` (or one of those `Assets` with enough `Versions`) to make these querysets large enough to blow up memory, but theoretically a malicious actor could create one like that and crash the worker instance.

@AlmightyYakob  @danlamanna any thoughts on this? Maybe this is just super-paranoia on my part from all the worker memory issues we had in the past due to iterating over large querysets w/out using `.iterator()`.

Reply:
I'm okay with it. I don't see any real downsides in doing so.